sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+   
+        #### Calculate Flux #####
+        # Use binned_statistic to determine which spectral bins contain
+        # components of line emission and sum over those line components
+        # to get the total emission is each spectral bin.
+        flux = stats.binned_statistic(line_spectrum_bins, split_line_intensities,
+                                      "sum", n_energy_bins, (0, n_energy_bins-1)).statistic
+    else:
+        flux = np.zeros((n_temperatures, n_energy_bins))
+    
+    # Scale flux by observer distance, emission measure and spectral bin width
+    # and put into correct units.
+    energy_bin_widths = (energy_edges_keV[1:] - energy_edges_keV[:-1]) * u.keV
+    flux = (flux * _LINE_GRID["intensity unit"] / energy_bin_widths)
+
+    return flux
+
+
+def _interpolate_continuum_intensities(data_grid, log10T_grid, energy_grid_keV, energy_keV, log10T):
+    # Determine valid range based on limits of intensity grid's spectral extent
+    # and the normalized temperature component of intensity.
+    n_tband = len(log10T_grid)
+    vrange, = np.where(data_grid[0] > 0)
+    for i in range(1, n_tband):
+        vrange_i, = np.where(data_grid[i] > 0)
+        if len(vrange) < len(vrange_i):
+            vrange = vrange_i
+    data_grid = data_grid[:, vrange]
+    energy_grid_keV = energy_grid_keV[vrange]
+    energy_idx, = np.where(energy_keV < energy_grid_keV.max())
+
+    # Interpolate temperature component of intensity and derive continuum intensity.
+    flux = np.zeros(energy_keV.shape)
+    if len(energy_idx) > 0:
+        energy_keV = energy_keV[energy_idx]
+        cont0 = interpolate.interp1d(energy_grid_keV, data_grid[0])(energy_keV)
+        cont1 = interpolate.interp1d(energy_grid_keV, data_grid[1])(energy_keV)
+        cont2 = interpolate.interp1d(energy_grid_keV, data_grid[2])(energy_keV)
+        # Calculate the continuum intensity as the weighted geometric mean
+        # of the interpolated values across the temperature band of the
+        # temperature component of intensity.
+        logelog10T = np.log(log10T)
+        x0, x1, x2 = np.log(log10T_grid)
+        flux[energy_idx]  = np.exp(
+            np.log(cont0) * (logelog10T - x1) * (logelog10T - x2) / ((x0 - x1) * (x0 - x2)) +
+            np.log(cont1) * (logelog10T - x0) * (logelog10T - x2) / ((x1 - x0) * (x1 - x2)) +
+            np.log(cont2) * (logelog10T - x0) * (logelog10T - x1) / ((x2 - x0) * (x2 - x1)) )
+    return flux
+
+
+def _calculate_abundance_normalized_line_intensities(logT, data_grid, line_logT_bins):
+    """
+    Calculates normalized line intensities at a given temperature using interpolation.
+
+    Given a 2D array, say of line intensities, as a function of two parameters,
+    say energy and log10(temperature), and a log10(temperature) value,
+    interpolate the line intensities over the temperature axis and
+    extract the intensities as a function of energy at the input temperature.
```

Comment:
```suggestion
    Given a user-defined log10 temperature and a 2D array of line intensities whose first axis
    corresponds to energy and second axis corresponds to log10(temperature),
    interpolate the line intensities over temperature and
    extract the intensities as a function of energy at the user-input temperature.
```
Is this clearer @hayesla?